sunpy/net/hek/utils.py

```diff
@@ -0,0 +1,242 @@
+import os
+import re
+import json
+import warnings
+from pathlib import Path
+
+import numpy.ma
+from regions import PolygonSkyRegion
+
+from astropy import units as u
+from astropy.coordinates import SkyCoord
+
+from sunpy.time import parse_time
+
+__all__ = [
+    'freeze',
+    'parse_times',
+    'parse_values_to_quantities',
+    'UNIT_FILE_PATH',
+    'COORD_FILE_PATH'
+]
+
+UNIT_FILE_PATH = Path(os.path.dirname(__file__)) / "unit_properties.json"
+COORD_FILE_PATH = Path(os.path.dirname(__file__)) / "coord_properties.json"
+
+def freeze(obj):
+    """
+    Create a hashable representation of a dict or list.
+    Otherwise returns the original input.
+    """
+    if isinstance(obj, dict):
+        return tuple((k, freeze(v)) for k, v in obj.items())
+    if isinstance(obj, list):
+        return tuple(freeze(elem) for elem in obj)
+    return obj
+
+# NOTE: Needs unit test
+def parse_times(table):
+    """
+    Parses the time values in an Astropy table into `astropy.time.Time`.
+
+    Parameters
+    ----------
+    table: `astropy.table`
+        Astropy table.
+
+    Returns
+    -------
+    `astropy.table`
+    """
+    # All time columns from https://www.lmsal.com/hek/VOEvent_Spec.html
+    time_keys = ['event_endtime', 'event_starttime', 'event_peaktime']
+    for tkey in time_keys:
+        if tkey in table.colnames:
+            table[tkey] = [
+                (parse_time(time, format='iso') if time else numpy.ma.masked) for time in table[tkey]
+            ]
+    return table
+
+# NOTE: Needs unit test
+def parse_values_to_quantities(table):
+    """
+    Parses the values in an Astropy table into Astropy objects.
+
+    Parameters
+    ----------
+    table: `astropy.table`
+        Astropy table.
+
+    Returns
+    -------
+    `astropy.table`
+
+    Raises
+    ------
+    TypeError
+        If `table` is not an Astropy table.
+    """
+    with open(UNIT_FILE_PATH) as unit_file:
+        unit_properties = json.load(unit_file)
+    unit_attributes = unit_properties["attributes"]
+
+    with open(COORD_FILE_PATH) as coord_file:
+        coord_properties = json.load(coord_file)
+    coord_attributes = coord_properties["attributes"]
+    table = parse_columns_to_table(table, unit_attributes)
+    table = parse_columns_to_table(table, coord_attributes, is_coord_prop = True)
+    return table
+
+# NOTE: Needs unit test
```

Comment:
This still true?

Reply:
Codecov says these functions are covered in the unit tests, but I didn't write tests for them, I don't know how

Reply:
Can you check the test files to see if there is one? If its not in test_utils, it should be moved.

Reply:
There is no test in `test_hek` that directly triggers one of them, maybe they are triggered indirectly by other tests?

Reply:
It is possible that as its used inside another function, that counts.
Tracing that with a search should be doable.

But it still should get a dedicated test(s).